jx_sqlite/insert_table.py

```diff
@@ -281,6 +282,28 @@ def _flatten(data, uid, parent_id, order, full_path, nested_path, row=None, guid
                 elif c.type=="nested" and value_type=="object":
                     value_type = "nested"
                     v = [v]
+                elif c.es_index!=table:
+                    from_doc = doc_collection.get(c.nested_path[0], None)
+                    column = c.es_column
+                    from_doc.active_columns.remove(c)
+                    abs_schema.remove(cname, c)
+                    required_changes.append({"nest": (c, nested_path[0])})                    
+                    c = Column(
+                        names={".": cname},
+                        type=value_type,
+                        es_column=typed_column(cname, value_type),
+                        es_index=table,
+                        nested_path=nested_path
+                    )
+                    abs_schema.add(cname, c)
+                    insertion.active_columns.add(c)
+
+                    for r in from_doc.rows:
```

Comment:
I am having difficulty understanding what you are doing here.  I suspect it is related to the fact that, at this point in the code, there are some records that have been _flattened, and must b nested.  I am not sure. You may tell me more, or I will try a bit more to figure it out.